components/event-component-factory.js

```diff
@@ -0,0 +1,109 @@
+// component-factory.js
+
+var _ = require('underscore');
+var noflo = require('noflo');
+
+/**
+ * Creates a noflo Component factory function from a component definitation,
+ * registers any event handlers on definitation. Triggers ondata events for outPorts.
```

Comment:
If I understand correctly, the purpose of triggering ondata events for outPorts is that it might provide a convenient mechanism for the Framework to invoke other functions in the framework.  But since I have not yet seen how the Framework would benefit (instead of calling another function directly), I do not yet feel convinced that we should keep it.  We can wait a bit and see though.